### ISSUE TITLE
Sorting order of index should ignore markup

### ISSUE THREAD

**Issue by gdamore** [daviddrysdale/mdbook-indexing#5]
I've noticed that the markup used impacts the sort order of terms in the index.

I want to use things like `NNG_DURATION` to make things a symbol, or italicize certain things even when they appear in the index.  This messes up the sorting order, which kind of defeats the purpose of the index.

**Comment by daviddrysdale** [comment]
The code does attempt to sort regardless of markup (e.g. the [*Effective Rust* index](https://effective-rust.com/indexing.html) intersperses code, italics, things like `'static` and `@name`s in a sensible sort order).

But there may well be a bug or something that's not coped with.  Can you give some specific examples that sort incorrectly?

**Comment by gdamore** [comment]
well I have an index that looks like this:

```
inproc transport, [1](http://localhost:3000/transports/inproc.html#a003)
PAIR‍, see protocol, PAIR
PUB protocol, [1](http://localhost:3000/protocols/pub.html#a001)
PULL protocol, [1](http://localhost:3000/protocols/pull.html#a001)
PUSH protocol, [1](http://localhost:3000/protocols/push.html#a001)
REP protocol, [1](http://localhost:3000/protocols/rep.html#a002)
REQ protocol, [1](http://localhost:3000/protocols/req.html#a001)
tcp transport, [1](http://localhost:3000/transports/tcp.html#a002)
ABI, [1](http://localhost:3000/api/util/nng_version.html#a005)
application binary interface, [1](http://localhost:3000/api/util/nng_version.html#a006)
asynchronous I/O, [1](http://localhost:3000/api/aio/nng_aio_alloc.html#a001), [2](http://localhost:3000/api/aio/index.html#a001)
```

**Comment by gdamore** [comment]
The copy paste didn't get it, but the entries above ABI all start italicized.

**Comment by gdamore** [comment]
I do see that options that are in backticks (`literal`) are sorted in the right order.  So maybe it's just italics that are not right?

**Comment by daviddrysdale** [comment]
Ah, I think I know what it is &ndash; I always use `*` for *italics*, but I'm guessing that you use _underscore_ (`_`) instead?

(I'll update the code that strips formatting to cope with either.)

**Comment by gdamore** [comment]
Thanks.

**Comment by daviddrysdale** [comment]
Fixed in 4b3fe9bcf6a0bc58ae71190c028af0ef98d41d01
